acme/acme/challenges.py

```diff
@@ -107,6 +107,7 @@ class KeyAuthorizationChallengeResponse(ChallengeResponse):
     """
     key_authorization = jose.Field("keyAuthorization")
     thumbprint_hash_function = hashes.SHA256
+    authorization_key_config = {'dump': False}
```

Comment:
This is a class level attribute so changing it affects all `KeyAuthorizationChallengeResponse` objects. I'm not sure if this is the behavior you wanted, but I think it's one step away from mutable global state which I'd personally like to avoid.

Can we make changes here so if there's any saved state in this object, it only affects the current instance? If we want to remember this setting for other objects, can we save the state in the `acme.client` object?

Reply:
It was not my intention indeed. I was a little puzzled by the behavior of `josepy` package, that change a lot of default Python object behavior, including the immutability, so I forgot to fix that.

I fix it now.

Reply:
If you spend much time fighting this, let me know and I'm happy to help you. `acme` and `josepy` have a lot of (largely unnecessary) magic going on.

Reply:
That is OK. The most difficult think was to understand that declaring a class parameter an assign it to the result of a josepy.Field result was creating effectively an instance parameter suitable for a controlled deserialization. And that most of the natural mutability of a python object was disabled in this case.